--- concordance-0.24/libconcord/Makefile.am.orig	2012-10-18 10:37:00.256518957 +0800
+++ concordance-0.24/libconcord/Makefile.am	2012-10-18 10:37:00.266519212 +0800
@@ -43,6 +43,17 @@
 	$(install_sh_DATA) libconcord.perms \
 		$(DESTDIR)$(datarootdir)/hal/fdi/policy/10osvendor/99-libconcord.perms
 
+udev_acl1:
+	./gen_udev_support -a
+
+install_udev_acl1 install_udev_acl2:
+	$(MKDIR_P) $(DESTDIR)/lib/udev/rules.d
+	$(install_sh_DATA) libconcord.rules \
+		$(DESTDIR)/lib/udev/rules.d/60-libconcord.rules
+
+udev_acl2:
+	./gen_udev_support -b
+
 # MIME stuff
 
 newmimedir = $(datadir)/mime/packages
--- concordance-0.24/libconcord/gen_udev_support.orig	2012-10-18 10:37:00.000000000 +0800
+++ concordance-0.24/libconcord/gen_udev_support	2012-10-18 10:40:22.361678889 +0800
@@ -17,6 +17,8 @@
 UDEV_POLICY_TEMPLATE='ATTR{idVendor}=="%s", ATTR{idProduct}=="%s", SYMLINK+="harmony-%%k"'
 UDEV_NEW_NO_POLICY_TEMPLATE='ATTR{idVendor}=="%s", ATTR{idProduct}=="%s", ENV{ID_REMOTE_CONTROL}="1"'
 UDEV_OLD_NO_POLICY_TEMPLATE='SYSFS{idVendor}=="%s", SYSFS{idProduct}=="%s", MODE="0660", GROUP="dialout"'
+UDEV_ACL1_TEMPLATE='ATTR{idVendor}=="%s", ATTR{idProduct}=="%s",ENV{ACL_MANAGE}="1"'
+UDEV_ACL2_TEMPLATE='ATTR{idVendor}=="%s", ATTR{idProduct}=="%s",TAG+="udev-acl"'
 
 HAL_PRE_TEMPLATE='    <match key="usb_device.vendor_id" int="0x%s">'
 HAL_RULE_TEMPLATE='      <match key="usb_device.product_id" int="0x%s">
@@ -83,6 +85,10 @@
     type="$2"
     if [ "$type" == 'policykit' -o "$type" == 'consolekit' ]; then
         template="$UDEV_POLICY_TEMPLATE"
+    elif [ "$type" == 'udev_acl1' ]; then
+        template="$UDEV_ACL1_TEMPLATE"
+    elif [ "$type" == 'udev_acl2' ]; then
+        template="$UDEV_ACL2_TEMPLATE"
     elif [ "$type" == 'old_udev' ]; then
         template="$UDEV_OLD_NO_POLICY_TEMPLATE"
     else
@@ -192,14 +198,14 @@
 #
 
 usage() {
-    echo "Usage: $0 <-u|-p|-c|-o>"
+    echo "Usage: $0 <-u|-p|-c|-o|-a|-b>"
 }
 
 #
 # MAIN
 #
 
-while getopts upco opt; do
+while getopts upcoab opt; do
     case $opt in
         u)
             MODE='udev_only'
@@ -213,6 +219,12 @@
         o)
             MODE='old_udev'
             ;;
+        a)
+            MODE='udev_acl1'
+            ;;
+        b)
+            MODE='udev_acl2'
+            ;;
         *)
             usage
             exit 1
