--- concordance-0.24/libconcord/gen_udev_support.orig	2012-03-15 15:03:25.000000000 +0800
+++ concordance-0.24/libconcord/gen_udev_support	2012-10-18 10:36:25.505632770 +0800
@@ -65,18 +65,15 @@
     file="$1"
     cat >$file <<END
 # Neat trick so that non-harmony devices don't read through a million rules
-SUBSYSTEM=="usb_device", GOTO="harmony_usb_rules"
-SUBSYSTEM=="usb", GOTO="harmony_usb_rules"
+ACTION!="add|change", GOTO="harmony_rules_end"
 SUBSYSTEM!="usb", GOTO="harmony_rules_end"
-GOTO="harmony_rules_end"
-LABEL="harmony_usb_rules"
+ENV{DEVTYPE}!="usb_device", GOTO="harmony_rules_end"
 END
 }
 
 emit_udev_footer() {
     file="$1"
     cat >>$file <<END
-GOTO="harmony_rules_end"
 LABEL="harmony_rules_end"
 END
 }
